src/Java.Interop.Tools.JavaSource/Java.Interop.Tools.JavaSource/SourceJavadocToXmldocGrammar.HtmlBnfTerms.cs

```diff
@@ -100,48 +99,53 @@ internal void CreateRules (SourceJavadocToXmldocGrammar grammar)
-							}
-						} else {
-							unparsedAElementValue += cn.AstNode?.ToString ();
-						}
-					}
+					var nodesAsString = GetChildNodesAsString (parseNode);
+					var tokenValue = parseNode.ChildNodes [0].Token.Text;
+					int stopIndex = nodesAsString.IndexOf ('>');
 
-					var seeElement = TryParseHRef (unparsedAElementValue);
-					if (seeElement == null)
-						seeElement = TryParseHRef (WebUtility.HtmlDecode (unparsedAElementValue), logError: true);
+					if (stopIndex == -1 || !tokenValue.Contains ("href", StringComparison.OrdinalIgnoreCase)) {
+						parseNode.AstNode = new XText (nodesAsString);
+						return;
+					}
 
-					var hrefValue = seeElement?.Attribute ("href")?.Value ?? string.Empty;
-					if (!string.IsNullOrEmpty (hrefValue) &&
-							(hrefValue.StartsWith ("http", StringComparison.OrdinalIgnoreCase) || hrefValue.StartsWith ("www", StringComparison.OrdinalIgnoreCase))) {
-						parseNode.AstNode = seeElement;
+					var attributeName = parseNode.ChildNodes [0].Term.Name;
+					var attributeValue = nodesAsString.Substring (0, stopIndex).Trim ().Trim('\'', '"');
+					var elementValue = nodesAsString.Substring (stopIndex + 1);
+					if (!string.IsNullOrEmpty (attributeValue) &&
+						(attributeValue.StartsWith ("http", StringComparison.OrdinalIgnoreCase) || attributeValue.StartsWith ("www", StringComparison.OrdinalIgnoreCase))) {
+						var unparsed = $"<see href=\"{attributeValue}\">{elementValue}</see>";
+						XNode? seeElement = TryParseElement (unparsed);
+						if (seeElement == null) {
+							// Try to parse with HTML entities decoded
+							seeElement = TryParseElement (WebUtility.HtmlDecode (unparsed));
+							if (seeElement == null) {
+								// Finally, try to parse with only the element value encoded
+								seeElement = TryParseElement ($"<see href=\"{attributeValue}\">{WebUtility.HtmlEncode (elementValue)}</see>", logError: true);
+							}
+						}
+						parseNode.AstNode = seeElement ?? new XText (nodesAsString);
 					} else {
 						// TODO: Need to convert relative paths or code references to appropriate CREF value.
-						parseNode.AstNode = new XText (unparsedAElementValue);
+						parseNode.AstNode = new XText (elementValue);
 					}
 				};
+			}
 
-				// Start to trim out unusable HTML elements/tags, but not any inner values
-				IgnorableElementDeclaration.Rule =
-					CreateStartElementIgnoreAttribute ("a", "name") + InlineDeclarations + CreateEndElement ("a", grammar, optional: true)
-					| CreateStartElementIgnoreAttribute ("a", "id") + InlineDeclarations + CreateEndElement ("a", grammar, optional: true)
-					;
-				IgnorableElementDeclaration.AstConfig.NodeCreator = (context, parseNode) => {
-					var aElementValue = new XText (parseNode.ChildNodes [1].AstNode.ToString () ?? string.Empty);
-					parseNode.AstNode = aElementValue;
-				};
-
-				CodeElementDeclaration.Rule = CreateStartElement ("code", grammar) + InlineDeclarations + CreateEndElement ("code", grammar);
```

Comment:
Alternate (terrible?) idea: what if instead of `<code/> + InlineDeclarations + </code>`, we instead used some Regex construct to just "slurp up" everything between `<code>` and `</code>`?  Maybe `new RegexBasedTerminal ("code", "<code>.*?</code>")`?

Reply:
I tried something similar, but didn't find anything that handled all of the corner cases mentioned above well. Another approach I played around with was adding an MSBuild task that ran before generator and would find/replace content in any javadoc.xml files that would be passed to generator. The task took a "replacements" file as a parameter that looked something like this:

```
<code>null</null>,,,,,<code>null</code>;;;;;
<code class=prettyprint>,,,,,<code>;;;;;
<code>Activity.RESULT_OK<code>,,,,,<code>Activity.RESULT_OK</code>;;;;;
...
```

The task would parse the `,,,,,` and `;;;;;` separators in the file and replace content in the provided javadoc files those keys/values. This way Irony wouldn't have to worry about these corner cases that are probably unique to the Mono.Android.dll use case. Maybe it's worth revisiting that?

Reply:
The regex suggestion is working well, and we can revisit the addition of a "javadoc fixup" build task in xamarin-android in the future if/when more corner cases arise.